<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1614</width>
    <height>905</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="CustomView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>831</width>
      <height>701</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="Graph" native="true">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>380</y>
      <width>731</width>
      <height>461</height>
     </rect>
    </property>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;规划路径：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>请输入起点：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>请输入终点：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="d_in1">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>70</y>
       <width>131</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="d_in2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>120</y>
       <width>131</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>30</y>
       <width>221</width>
       <height>151</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>最短距离为：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="d1">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>最少用时为：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="d2">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="mind">
     <property name="geometry">
      <rect>
       <x>550</x>
       <y>30</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>查看最短路径</string>
     </property>
    </widget>
    <widget class="QPushButton" name="mint">
     <property name="geometry">
      <rect>
       <x>550</x>
       <y>120</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>查看最优路径</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget_4">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>270</y>
       <width>371</width>
       <height>51</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <item>
       <widget class="QLabel" name="label_20">
        <property name="text">
         <string>最块路线：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="tout">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="d_start">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>340</y>
       <width>181</width>
       <height>71</height>
      </rect>
     </property>
     <property name="text">
      <string>开始</string>
     </property>
    </widget>
    <widget class="QPushButton" name="d_end">
     <property name="geometry">
      <rect>
       <x>390</x>
       <y>340</y>
       <width>201</width>
       <height>71</height>
      </rect>
     </property>
     <property name="text">
      <string>结束</string>
     </property>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>200</y>
       <width>371</width>
       <height>51</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>最短路线：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="dout">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>0</y>
      <width>371</width>
      <height>271</height>
     </rect>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>0</y>
       <width>191</width>
       <height>81</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;模拟车流：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QPushButton" name="m_start">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>200</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>开始</string>
     </property>
    </widget>
    <widget class="QPushButton" name="m_end">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>200</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>结束</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>60</y>
       <width>199</width>
       <height>54</height>
      </rect>
     </property>
     <property name="text">
      <string>请选择要显示的路段：</string>
     </property>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>110</y>
       <width>141</width>
       <height>81</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QCheckBox" name="m1">
        <property name="text">
         <string>通畅路段</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m2">
        <property name="text">
         <string>略堵路段</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="m3">
        <property name="text">
         <string>拥堵路段</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="widget_2" native="true">
    <property name="geometry">
     <rect>
      <x>1170</x>
      <y>0</y>
      <width>441</width>
      <height>391</height>
     </rect>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>20</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;寻找最近30个点&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>120</y>
       <width>301</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QPushButton" name="f_start">
        <property name="text">
         <string>开始</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="clear">
        <property name="text">
         <string>清除</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>200</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>最近的点是：</string>
     </property>
    </widget>
    <widget class="QLabel" name="out">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>230</y>
       <width>251</width>
       <height>111</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>70</y>
       <width>269</width>
       <height>41</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>请输入点号：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="f_in">
        <property name="inputMask">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="start">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>750</y>
      <width>141</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>生成图</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1614</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>导航系统</string>
    </property>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomView</class>
   <extends>QGraphicsView</extends>
   <header location="global">customview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
